File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/RamLfeProgramPolicyListResponse.java
package org.hyperledger.iroha.android.client;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Objects;

/** Immutable response wrapper for `GET /v1/ram-lfe/program-policies`. */
public final class RamLfeProgramPolicyListResponse {
  private final long total;
  private final List<RamLfeProgramPolicySummary> items;

  public RamLfeProgramPolicyListResponse(
      final long total, final List<RamLfeProgramPolicySummary> items) {
    this.total = total;
    this.items =
        Collections.unmodifiableList(new ArrayList<>(Objects.requireNonNull(items, "items")));
  }

  public long total() {
    return total;
  }

  public List<RamLfeProgramPolicySummary> items() {
    return items;
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/RamLfeProgramPolicyListResponse.kt
package org.hyperledger.iroha.sdk.client

/** Immutable response wrapper for `GET /v1/ram-lfe/program-policies`. */
class RamLfeProgramPolicyListResponse(
    @JvmField val total: Long,
    items: List<RamLfeProgramPolicySummary>,
) {
    @JvmField
    val items: List<RamLfeProgramPolicySummary> = items.toList()
}
